pkg/testutils/network/network.go

```diff
@@ -0,0 +1,35 @@
+// Package network implements the network test utils.
+package network
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func OpenListener(t *testing.T) net.Listener {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		_ = ln.Close()
+	})
+
+	return ln
+}
+
+// OpenFreePort opens a free port on the local machine.
+// It should be used only in tests, where an http server
+// needs a port, but unlike gRPC, cannot reuse a listener.
+func OpenFreePort(t *testing.T) int {
```

Comment:
this does not work. It will open a port and then immediately close it. That port does not remain open and another process can sneak it. If you want to do this, you must return the open socket.